main/inc/lib/exercise.lib.php

```diff
@@ -3795,27 +3795,56 @@ public static function get_exercises_to_be_taken($course_code, $session_id)
      * @param int    $exercise_id
      * @param string $course_code
      * @param int    $session_id
+     * @param bool   $onlyStudent Filter only enrolled students
      *
      * @return array
      */
     public static function get_student_stats_by_question(
         $question_id,
         $exercise_id,
         $course_code,
-        $session_id
+        $session_id,
+        $onlyStudent = false
     ) {
         $track_exercises = Database::get_main_table(TABLE_STATISTIC_TRACK_E_EXERCISES);
         $track_attempt = Database::get_main_table(TABLE_STATISTIC_TRACK_E_ATTEMPT);
+        $courseUser = Database::get_main_table(TABLE_MAIN_COURSE_USER);
 
         $question_id = (int) $question_id;
         $exercise_id = (int) $exercise_id;
         $course_code = Database::escape_string($course_code);
         $session_id = (int) $session_id;
         $courseId = api_get_course_int_id($course_code);
 
+        if (empty($session_id)) {
+            $courseCondition = "
+            INNER JOIN $courseUser c
+            ON (
+                        e.exe_user_id = c.user_id AND
+                        e.c_id = c.c_id AND
+                        c.status = ".STUDENT."
+                        AND relation_type <> 2
+                )
+            ";
+        } else {
+            $courseCondition = "
+            INNER JOIN $courseUser c
+            ON (
+                        e.exe_user_id = c.user_id AND
+                        e.c_id = c.c_id AND
+                        c.status = 0
+                )
+            ";
+        }
+
         $sql = "SELECT MAX(marks) as max, MIN(marks) as min, AVG(marks) as average
     		FROM $track_exercises e
-    		INNER JOIN $track_attempt a
+    		";
+        if ($onlyStudent == true) {
+            $sql .= $courseCondition;
```

Comment:
If you only use $courseCondition *if* $onlyStudent is true, then it shouldn't be necessary to build $courseCondition if $onlyStudent is not true (so you should build it *inside* the `if ($onlyStudent == true)` block.
I will do it myself to avoid delays, but I wanted to make sure you see it.